BrainPortal/lib/boutiques_input_value_fixer.rb

```diff
@@ -0,0 +1,192 @@
+# still able to enter correct input values by ignoring -max_mem_Gig and -all_mem. This particular cases is addressed
+# by this module by deletion of unfixed elements of the group. However, generally speaking, the module
+# does not always guaranty that in the transformed descriptor
+# would allow parameter combinations, which are prevented by previous descriptor
+#
+# Example 2 (A more severe case of implied dependencies)
+# # change is possible. Let consider a case when the above boutiques has also 'one-is-required' flag for the same group.
+# In this case fixing value of one parameter while keeping one-is-required and flag, the
+#
+# Fixer aims at avoiding second case of the stopping user from entering inputs, possible with given choice of fixed inputs
+# yet does try hard at blocking all the combos impossible in the original boutiques.
+module BoutiquesInputValueFixer
+
+  # Note: to access the revision info of the module,
+  # you need to access the constant directly, the
+  # object method revision_info() won't work.
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+  # the hash of param values to be fixed or be ommited ()  #
+  def fixed_values
+    self.boutiques_descriptor.custom_module_info('BoutiquesInputValueFixer')
+  end
+
+  require 'pry'
+
+  # deletes fixed inputs listed in the custom 'integrator_modules'
+  def descriptor_without_fixed_inputs(descriptor)
+    # input parameters are marked by null values will be excluded from the command line
+    # the major use case are Flags, but also may be useful to address params with 'default'
+
+    descriptor_dup = descriptor.dup
+    todelete = fixed_values.keys.select do |i_id|  # this variables are flagged to be removed rather than assigned value
+                                                   # in the spec, so they will be treated slightly different
+      input = descriptor_dup.input_by_id(i_id)
+      value = fixed_values[i_id]
+      value.nil? || (input.type == 'Flag') &&  (value.presence.to_s.strip =~ /no|0|nil|none|null|false/i || value.blank?)
+
+    end
+
+    # generally speaking, boutiques input groups can have three different constraints,
+    # here we address mutually exclusion group only, which is the only one present in dynamic GUI (rest are evaluated
+    # after submission of parameter).
+
+    descriptor_dup.groups.each do |g| # filter groups, relax restriction to ensure that form can still be submitted
+      members = g.members - fixed_values.keys
+      # disable a mutualy exclusive group if its param assigned fixed value by this modifier
+      # if one simply deletes the fixed param,
+      if g.try(:mutually_exclusive) && members.length != g.members.length # params can be mutually exclusive e.g. --use-min-mem vs --mem-mb
+        if (fixed_values.keys & g.members - todelete).present? # at least some group members are actually assigned vals rather than deleted
+          g.mutually_exclusive = false
+          if g.try(:one_is_required).present?
```

Comment:
Why is there a `try` here? `g` is always not null, so you will always invoke `g.one_is_required`